reasoner_pydantic/qgraph.py

```diff
@@ -114,7 +114,6 @@ class QEdge(BaseModel):
     _nonzero_predicates = validator("predicates", allow_reuse=True)(nonzero_validator)
 
     constraints: Optional[HashableSequence[QueryConstraint]] = Field(
-        [],
```

Comment:
The first value of the Field function is a default value. Pydantic loads default values directly into __root__ so we end up with a __root__ that is an empty list instead of a set.